components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -420,100 +406,109 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
-          constraintEqualToAnchor:self.containerView.leadingAnchor],
-      [self.safeAreaLayoutGuide.topAnchor constraintEqualToAnchor:self.containerView.topAnchor],
-      [self.safeAreaLayoutGuide.trailingAnchor
-          constraintEqualToAnchor:self.containerView.trailingAnchor],
-      [self.safeAreaLayoutGuide.bottomAnchor
-          constraintEqualToAnchor:self.containerView.bottomAnchor],
-    ];
-    self.scrollableLayoutConstraints = @[
-      [self.contentLayoutGuide.topAnchor constraintEqualToAnchor:self.containerView.topAnchor],
-      [self.contentLayoutGuide.bottomAnchor
-          constraintEqualToAnchor:self.containerView.bottomAnchor],
-      [self.contentLayoutGuide.leadingAnchor
-          constraintEqualToAnchor:self.containerView.leadingAnchor],
-      [self.contentLayoutGuide.trailingAnchor
-          constraintEqualToAnchor:self.containerView.trailingAnchor],
-      [self.contentLayoutGuide.widthAnchor constraintEqualToAnchor:self.containerView.widthAnchor],
-      [self.contentLayoutGuide.heightAnchor
-          constraintEqualToAnchor:self.containerView.heightAnchor],
-      // Ensures items are never larger than the bar.
-      [self.frameLayoutGuide.heightAnchor
-          constraintGreaterThanOrEqualToAnchor:self.containerView.heightAnchor],
-    ];
-  } else {
-    self.justifiedLayoutConstraints = @[
-      [self.heightAnchor constraintEqualToAnchor:self.containerView.heightAnchor],
-      [self.widthAnchor constraintEqualToAnchor:self.containerView.widthAnchor],
-    ];
-    self.scrollableLayoutConstraints = @[];
-    [self.leadingAnchor constraintEqualToAnchor:self.containerView.leadingAnchor].active = YES;
-    [self.topAnchor constraintEqualToAnchor:self.containerView.topAnchor].active = YES;
-    [self.trailingAnchor constraintEqualToAnchor:self.containerView.trailingAnchor].active = YES;
-    [self.bottomAnchor constraintEqualToAnchor:self.containerView.bottomAnchor].active = YES;
+    availableBounds = UIEdgeInsetsInsetRect(availableBounds, self.adjustedContentInset);
   }
-  self.containerViewConstraintsActive = YES;
+  BOOL isRTL =
+      [self mdf_effectiveUserInterfaceLayoutDirection] == UIUserInterfaceLayoutDirectionRightToLeft;
 
-  // Must always be called last according to the documentation.
-  [super updateConstraints];
+  CGFloat itemViewOriginX = 0;
+  CGFloat itemViewOriginY = 0;
+  CGFloat itemViewHeight = CGRectGetHeight(availableBounds);
+  NSEnumerator<UIView *> *itemViewEnumerator =
+      isRTL ? [self.itemViews reverseObjectEnumerator] : [self.itemViews objectEnumerator];
+  for (UIView *view in itemViewEnumerator) {
+    CGSize intrinsicContentSize = view.intrinsicContentSize;
+    view.frame =
+        CGRectMake(itemViewOriginX, itemViewOriginY, intrinsicContentSize.width, itemViewHeight);
+    itemViewOriginX += intrinsicContentSize.width;
+  }
+}
+
+- (void)willMoveToSuperview:(UIView *)newSuperview {
+  [super willMoveToSuperview:newSuperview];
+  self.initialScrollDone = NO;
```

Comment:
Consider changing this to “needsScrollToSelectedItem”

Reply:
Done